simpeg/flow/richards/empirical.py

```diff
@@ -34,12 +36,19 @@ class NonLinearModel(props.HasModel):
     """A non linear model that has dependence on the fields and a model"""
 
     counter = None  #: A simpeg.utils.Counter object
-    mesh = None  #: A discretize Mesh
 
     def __init__(self, mesh, **kwargs):
         self.mesh = mesh
         super(NonLinearModel, self).__init__(**kwargs)
 
+    @property
```

Comment:
I did this because it already accepted a mesh argument, but I don't think this is ever used anywhere, and this base class doesn't really do anything anyways and should likely just be removed.